### ISSUE TITLE
Rule Request: force_exhaustive_switch

### ISSUE THREAD

**Issue by superarts** [realm/SwiftLint#3728]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request `force_exhaustive_switch`

Enable to force exhaustive switch unless bypassed.

There's [an ancient issue about this](https://github.com/realm/SwiftLint/issues/131) which stated exhaustive switch was very helpful. It is so useful that in our case, we would like to make it mandatory by not allowing `default:` clause.

No configuration is needed. And it should be disabled by default as no one was requesting it before, so there must be not enough people who really want it, as it's more like a choice of style.

However, in our case it's more than that. We use Swiftlint to fail our CI and want it to be configured it in a way that unless developer explicitly `bypass` this rule, we will fail CI and force developers to use exhaustive switch. This is especially useful for things like managing server configuration etc. so that when something new is added, it's hard to miss something. So:

3. Should the rule be configurable, if so what parameters should be configurable? - NO
4. Should the rule be opt-in or enabled by default? Why? - NO

#### What would trigger

```swift
switch myEnum {
case .case1: ()
default: ()
}
```

   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.
